spectec/src/frontend/multiplicity.ml

```diff
@@ -290,6 +285,12 @@ and annot_iterexp env occur1 (iter, ids) at : Il.Ast.iterexp * occur =
     ) occur1
   in
   let ids' = List.map (fun (x, _) -> x $ at) (Env.bindings occur1') in
+
+  (match iter' with
+  | ListN (_, Some id) ->
+    assert (List.length ids' = 1 && id.it = (List.hd ids').it)
```

Comment:
It is because I would like to block the use of "iter with index" like this:
```
def $foo() : nat
def $foo() = 0
  -- if x* = 0 1 2
  -- if (x*[i] = y)^(i<3)
```
If you think we should allow semantics like this, I will remove the assertion.

Reply:
I don't think I follow, what problem do you see with that example? In Shouldn't using them for indexing be one of the primary use cases for iter vars?

Reply:
I apologize for not providing sufficient explanation earlier.
My intention was to use 'iter with index' in a way that restricts iteration on variables other than the index variable.
However, upon further consideration, I realized that 'iter with index' merely introduces an index for iteration, so there shouldn't be such a restriction. I'll erase the assertion.